sqlmesh/core/test/__init__.py

```diff
@@ -43,46 +54,107 @@ def run_tests(
     testing_adapter_by_gateway: t.Dict[str, EngineAdapter] = {}
     default_gateway = gateway or config.default_gateway_name
 
-    try:
-        tests = []
-        for metadata in model_test_metadata:
-            body = metadata.body
-            gateway = body.get("gateway") or default_gateway
-            testing_engine_adapter = testing_adapter_by_gateway.get(gateway)
-            if not testing_engine_adapter:
-                testing_engine_adapter = config.get_test_connection(
-                    gateway,
-                    default_catalog,
-                    default_catalog_dialect,
-                ).create_engine_adapter(register_comments_override=False)
-                testing_adapter_by_gateway[gateway] = testing_engine_adapter
-
-            test = ModelTest.create_test(
-                body=body,
-                test_name=metadata.test_name,
-                models=models,
-                engine_adapter=testing_engine_adapter,
-                dialect=dialect,
-                path=metadata.path,
-                default_catalog=default_catalog,
-                preserve_fixtures=preserve_fixtures,
+    default_test_connection = config.get_test_connection(
+        gateway_name=default_gateway,
+        default_catalog=default_catalog,
+        default_catalog_dialect=default_catalog_dialect,
+    )
+
+    lock = threading.Lock()
+
+    combined_results = ModelTextTestResult(
+        stream=unittest.runner._WritelnDecorator(stream or sys.stderr),  # type: ignore
+        verbosity=2 if verbosity >= Verbosity.VERBOSE else 1,
+        descriptions=None,
+    )
+
+    worker_payload = []
```

Comment:
Don't feel strongly about this, but I think I'd consider extracting the logic that follows out into a helper function that constructs the worker payloads. Right now, `run_tests` spans L34-L157 and it feels a bit loaded because it does various things.

Also, a nit, but I'd rename to plural form.